### DISCUSSION TITLE
Prebuilt install not detecting

### DISCUSSION THREAD

**Discussion by Sourabh25495** [TimelordUK/node-sqlserver-v8#342] [General]
Hi,

I am trying to install msnodesqlv8. I got it to work just fine by manually placing the prebuilts in node_modules/msnodesqlv8/build/Release. 

The process I follow is  i have the binaries checked in my code and in postinstall, I do npm install msnodesqlv8 --ignore-scripts. Then I copy the files to the specified location in node modules(build/Release). It works just fine.

The problem I am having is, if I do a seed folder in the node_modules(msnodesqlv8/build/Release) before the install and place the prebuilt binaries in it and then do a npm install msnodesqlv8 (without the --ignore-scripts) it doesn't detect that the binaries are already present there and tries to compile using node-gyp and look for headers(which I can't connect to). Note: I can't download the headers since I am behind the firewall.

Am I missing something because of which it wouldn't detect the seeded binary folder?

Thanks

**Comment by TimelordUK** [comment]
the library uses prebuild to search for a binary and install it, firstly it will check in the cache

if you run the following you should be able to see where the library will be pulled from if it is locally installed, my advice first is to manually copy the binary here.  If placed properly, prebuild should choose this location rather than trying to go externally which will cause problems as you say if you are behing a firewall.

```sh
npm run install-verbose
```

for example running locally, see the location from output below

```sh
 /home/me/.npm/_prebuilds/c3e42f-msnodesqlv8-v4.2.1-node-v127-linux-x64.tar.gz
```


```sh
prebuild-install info begin Prebuild-install version 7.1.2
prebuild-install info looking for local prebuild @ prebuilds/msnodesqlv8-v4.2.1-node-v127-linux-x64.tar.gz
prebuild-install info looking for cached prebuild @ /home/me/.npm/_prebuilds/c3e42f-msnodesqlv8-v4.2.1-node-v127-linux-x64.tar.gz
prebuild-install info found cached prebuild
prebuild-install info unpacking @ /home/me/.npm/_prebuilds/c3e42f-msnodesqlv8-v4.2.1-node-v127-linux-x64.tar.gz
prebuild-install info unpack resolved to /home/me/dev/js/sql/v8/node_modules/msnodesqlv8/build/Release/sqlserverv8.node
prebuild-install info unpack required /home/me/dev/js/sql/v8/node_modules/msnodesqlv8/build/Release/sqlserverv8.node successfully
prebuild-install info install Successfully installed prebuilt binary!
```
